### ISSUE TITLE
Pub get should give an actionable error message with SDK mismatch

### ISSUE THREAD

**Issue by matanlurey** [dart-lang/pub#1584]
Errors like:

```
Resolving dependencies...
Package angular2 has no versions that match >=3.0.0 <4.0.0 derived from
```

Sometimes are due to the installed SDK being too old. We should surface that clearly.

**Comment by zoechi** [comment]
seems to be a dup of #1318 and therefore #912

**Comment by matanlurey** [comment]
Maybe, but I think this should stand out as this is the user-visible error.

**Comment by nex3** [comment]
Closing as a duplicate. If we kept open a separate issue for every user-visible negative effect stemming from #912, they'd take up half the issue tracker :confused:.

**Comment by matanlurey** [comment]
This is one the disproportionally effects AngularDart right now since we frequently have a dependency on a bleeding SDK (due to new language features/analyzer). So stuffing this under issues in #912 which haven't gotten touched in a long time sort of mutes a really bad experience for Dart web users.

**Comment by nex3** [comment]
I sympathize, I really do—I know that the current error messages are awful. The unfortunate fact, though, is that there's not a good way to solve individual cases without solving the underlying problem that the version solver doesn't track enough information to construct a coherent explanation of what went wrong.

I spent a bunch of time working on this last year, and I made a lot of progress. I have a plan that I think will address this issue and improve version solver performance to boot. I've been pushing for allocating more of my time to finish that work, but so far I've been unsuccessful. I'm hoping to get back to it at some point this year.
